### ISSUE TITLE
html_permalinks_icon is not compatible with Sphinx 7.3

### ISSUE THREAD

**Issue by dolfinus** [pradyunsg/furo#787]
### What's happening?

After upgrading to Sphinx 7.3 I cannot build documentation:
```
Exception occurred:
  File "/home/maxim/Repo/onetl/venv/lib/python3.9/site-packages/sphinx/config.py", line 434, in __iter__
    yield ConfigValue(name, getattr(self, name), opt.rebuild)
AttributeError: 'tuple' object has no attribute 'rebuild'
```
https://readthedocs.org/projects/onetl/builds/24169241/

If I print `name` and `opt` values, they look like this:
```
html_additional_pages _Opt(default={}, rebuild='html', valid_types=())
html_domain_indices _Opt(default=True, rebuild='html', valid_types=frozenset({<class 'list'>}))
html_permalinks _Opt(default=True, rebuild='html', valid_types=())
html_permalinks_icon ('#', 'html', ())
```

This is there this option is being set:
https://github.com/pradyunsg/furo/blob/035b27651632e20fec2277e7398cdd00dc63a73f/src/furo/__init__.py#L286

Since Sphinx 7.3.0 tuple was replaced with a class:
https://github.com/sphinx-doc/sphinx/commit/fd23cf0256bd401211ae3535206f48f7e28bca64
And passing a tuple here breaks this code.

### Reproducer

* Install these packages:
```requirements.txt
furo==2024.1.29
sphinx==7.3.7
autodoc_pydantic==2.1.0
```
* Create `docs/conf.py` with content:
```python
extensions = [
    "sphinxcontrib.autodoc_pydantic",
]
html_theme = "furo"
```
* Add source code and documentation page:
`mymodel.py`
```python
from pydantic import BaseModel

class MyModel(BaseModel):
  """Some docstring"""
  abc: int
```

`docs/index.rst`
```rst
.. currentmodule:: mymodel

.. autopydantic_model:: MyModel
```
* Build documentation:
```bash
cd docs
make html
```

### Expectation

Documentations is build without any errors.

### Code of Conduct

- [X] I agree to follow the [Code of Conduct](https://github.com/pradyunsg/furo/blob/main/CODE_OF_CONDUCT.md).

**Comment by dolfinus** [comment]
Fix: #783

**Comment by dolfinus** [comment]
Fixed in https://github.com/pradyunsg/furo/releases/tag/2024.04.27
